### PR TITLE
[CloudPC]Add partner agent installation status and graph api for retry partner agent

### PR DESCRIPTION
**Pull request by jannyma**

### CODE REVIEW

**Comment by AshleyYangSZ** [comment]
@jannyma Please add the change-log as well.

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+doc_type: resourcePageType
+---
+
+# cloudPcPartnerAgentInstallResult resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Describe the detail of one partner agent installation result on a cloud pc.
```
Comment (RuiHou105) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:16]:
```suggestion
Describe installation result details for a partner agent on a Cloud PC.
```

File: api-reference/beta/api/cloudpc-retrypartneragentinstallation.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+``` http
+POST /deviceManagement/virtualEndpoint/cloudPCs/{id}/retryPartnerAgentInstallation
+```
+
+## Request headers
+
+|Name|Description|
+|:---|:---|
+|Authorization|Bearer {token}. Required.|
+|Content-Type|application/json. Required.|
```
Comment (RuiHou105) [inline api-reference/beta/api/cloudpc-retrypartneragentinstallation.md:44]: Remove this line as this action doesn't pass any json body to backend.

File: api-reference/beta/api/cloudpc-retrypartneragentinstallation.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  "blockType": "request",
+  "name": "cloudpc_retrypartneragentinstallation"
+}
+-->
+
+``` http
+POST https://graph.microsoft.com/beta/deviceManagement/virtualEndpoint/cloudPCs/831dd62e-cfa1-4d49-a3b4-58d4e9920f8e/retryPartnerAgentInstallation
+```
+
+# [C#](#tab/csharp)
```
Comment (RuiHou105) [inline api-reference/beta/api/cloudpc-retrypartneragentinstallation.md:71]: Please remove this part line 71 - 93. As these scripts will be automatically generated by tool after this PR gets merged. We don't need to add them manually.

**Review by RuiHou105 — CHANGES_REQUESTED** [review]
@jannyma 
1. please resolve comments in this PR.
2. Please also update the changelog/Microsoft.CloudManagedDesktop.json file later.
3. Please create a roadmap workitem and attach the workitem number in comments. Here is an [example](https://microsoftgraph.visualstudio.com/Roadmap/_workitems/edit/14460/). If no access, please do these two steps firstly.
- Request access via [Improved ADO Access - 1ES Wikirequest to join](https://www.1eswiki.com/wiki/Improved_ADO_Access) .
- Join the Microsoft Graph Roadmap Users ([graphroadmap@microsoft.com](mailto:graphroadmap@microsoft.com)) security group on [idweb](https://idwebelements/GroupManagement.aspx?Group=graphroadmap&Operation=join) .

**Comment by Danielabom** [comment]
Reached out to Jingyi @jannyma to get started.

File: api-reference/beta/api/cloudpc-retrypartneragentinstallation.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
+---
+title: "cloudPC: retryPartnerAgentInstallation"
+description: "Attempt to re-install party partner agents in a Cloud PC which were failed to install."
```
Comment (Danielabom) [inline api-reference/beta/api/cloudpc-retrypartneragentinstallation.md:3]:
```suggestion
description: "Attempts to re-install party partner agents in a Cloud PC failed to install."
```

File: api-reference/beta/api/cloudpc-retrypartneragentinstallation.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+doc_type: apiPageType
+---
+
+# cloudPC: retryPartnerAgentInstallation
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Attempt to re-install party partner agents in a Cloud PC which were failed to install.
```
Comment (Danielabom) [inline api-reference/beta/api/cloudpc-retrypartneragentinstallation.md:16]:
```suggestion
Attempts to re-install party partner agents in a Cloud PC failed to install.
```

**Comment by jannyma** [comment]
Roadmap work item no: 15432
https://microsoftgraph.visualstudio.com/Roadmap/_workitems/edit/15432

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
+---
+title: "cloudPcPartnerAgentInstallResult resource type"
+description: "Describe installation result details for a partner agent on a Cloud PC."
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:3]:
```suggestion
description: "Provides installation results for a partner agent on a Cloud PC."
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+author: "JannyMa"
+ms.localizationpriority: medium
+ms.prod: "cloud-pc"
+doc_type: resourcePageType
+---
+
+# cloudPcPartnerAgentInstallResult resource type
+
+Namespace: microsoft.graph
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:12]:
```suggestion
Namespace: microsoft.graph
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Describe installation result details for a partner agent on a Cloud PC.
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:16]:
```suggestion
Provides installation results for a partner agent on a Cloud PC.
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+## Properties
+
+|Property|Type|Description|
+|:---|:---|:---|
+|partnerAgentName|[cloudPcPartnerAgentName](#cloudpcpartneragentname-values)|Indicates the name of a partner agent, right now it's just Citrix, which is third-party. And it includes first-party and third-party. Read-Only.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:22]:
```suggestion
|partnerAgentName|[cloudPcPartnerAgentName](#cloudpcpartneragentname-values)|Indicates the name of a partner agent and includes first-party and third-party. Currently, Citrix is the only third-party value. Read-Only.|
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+|installStatus|[cloudPcPartnerAgentInstallStatus](#cloudpcpartneragentinstallstatus-values)|Indicates the status of a partner agent installation, possible value are: installed, installFailed, installing, uninstalling, uninstallFailed, licensed. Read-Only.|
+|isThirdPartyPartner|Boolean|Indicates if the partner agent is third-party. When 'TRUE', the agent is a third-party (non-Microsoft) agent.  When 'FALSE', the agent is a Microsoft agent or is not known.  Default value is 'FALSE'.|
+|retriable|Boolean|Indicates whether the partner agent installation should be retried. If the installStatus is 'installFailed' or 'licensed' then 'TRUE' indicates the installation should be retried.  If the installStatus is neither 'installFailed'nor 'licensed' then the installation will not be retried if the value is 'FALSE'. Default value is 'FALSE'.|
+
+### cloudPcPartnerAgentName values
+
+|Member|Description|
+|:---|:---|
+|citrix|Indicates that a third-party-partner named Citrix.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:31]:
```suggestion
|citrix|Indicates a third-party-partner named Citrix.|
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Describe installation result details for a partner agent on a Cloud PC.
+
+## Properties
+
+|Property|Type|Description|
+|:---|:---|:---|
+|partnerAgentName|[cloudPcPartnerAgentName](#cloudpcpartneragentname-values)|Indicates the name of a partner agent, right now it's just Citrix, which is third-party. And it includes first-party and third-party. Read-Only.|
+|installStatus|[cloudPcPartnerAgentInstallStatus](#cloudpcpartneragentinstallstatus-values)|Indicates the status of a partner agent installation, possible value are: installed, installFailed, installing, uninstalling, uninstallFailed, licensed. Read-Only.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:23]:
```suggestion
|installStatus|[cloudPcPartnerAgentInstallStatus](#cloudpcpartneragentinstallstatus-values)|Indicates the status of a partner agent installation. Possible values are: installed, installFailed, installing, uninstalling, uninstallFailed, and licensed. Read-Only.|
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+|isThirdPartyPartner|Boolean|Indicates if the partner agent is third-party. When 'TRUE', the agent is a third-party (non-Microsoft) agent.  When 'FALSE', the agent is a Microsoft agent or is not known.  Default value is 'FALSE'.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:24]:
```suggestion
|isThirdPartyPartner|Boolean|Indicates if the partner agent is a third party. When 'TRUE', the agent is a third-party (non-Microsoft) agent. When 'FALSE', the agent is a Microsoft agent or is not known. The default value is 'FALSE'.|
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+|retriable|Boolean|Indicates whether the partner agent installation should be retried. If the installStatus is 'installFailed' or 'licensed' then 'TRUE' indicates the installation should be retried.  If the installStatus is neither 'installFailed'nor 'licensed' then the installation will not be retried if the value is 'FALSE'. Default value is 'FALSE'.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:25]:
```suggestion
|retriable|Boolean|Indicates if the partner agent is a third party. When 'TRUE', the agent is a third-party (non-Microsoft) agent. When 'FALSE', the agent is a Microsoft agent or is not known. The default value is 'FALSE'.|
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+|Member|Description|
+|:---|:---|
+|installed|Indicates the partner agent has been installed.|
+|installFailed|Indicates the partner agent installation failed although the license has been assigned.|
+|installing|Indicates the partner agent is installing.|
+|uninstalling|Indicates that the partner agent will automatically be removed.  This occurs when there is no third-party-partner license but the third-party-partner agent was previously installed.|
+|uninstallFailed|Indicates the partner agent uninstallation failed although the license has already unassigned.|
+|licensed|Indicates the partner license was assigned to this Cloud PC, but the third-party-partner agent is not installed yet.|
+|unknownFutureValue|volvable enumeration sentinel value. Do not use.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:44]:
```suggestion
|unknownFutureValue|Evolvable enumeration sentinel value. Do not use.|
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+|:---|:---|
+|citrix|Indicates that a third-party-partner named Citrix.|
+|unknownFutureValue|Evolvable enumeration sentinel value. Do not use.|
+
+#### cloudPcPartnerAgentInstallStatus values
+
+|Member|Description|
+|:---|:---|
+|installed|Indicates the partner agent has been installed.|
+|installFailed|Indicates the partner agent installation failed although the license has been assigned.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:39]:
```suggestion
|installFailed|Indicates the partner agent installation failed, although a license was assigned.
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+|installing|Indicates the partner agent is installing.|
+|uninstalling|Indicates that the partner agent will automatically be removed.  This occurs when there is no third-party-partner license but the third-party-partner agent was previously installed.|
+|uninstallFailed|Indicates the partner agent uninstallation failed although the license has already unassigned.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:42]:
```suggestion
|uninstallFailed|Indicates the partner agent uninstallation failed, although a license was already unassigned.|
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+|licensed|Indicates the partner license was assigned to this Cloud PC, but the third-party-partner agent is not installed yet.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:43]:
```suggestion
|licensed|The partner license was assigned to this Cloud PC, but the third-party partner agent is not yet installed.|
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+|Member|Description|
+|:---|:---|
+|citrix|Indicates that a third-party-partner named Citrix.|
+|unknownFutureValue|Evolvable enumeration sentinel value. Do not use.|
+
+#### cloudPcPartnerAgentInstallStatus values
+
+|Member|Description|
+|:---|:---|
+|installed|Indicates the partner agent has been installed.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:38]:
```suggestion
|installed|The partner agent has been installed.|
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+|installFailed|Indicates the partner agent installation failed although the license has been assigned.|
+|installing|Indicates the partner agent is installing.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:40]:
```suggestion
|installing|The partner agent is installing.|
```

File: api-reference/beta/resources/cloudpcpartneragentinstallresult.md [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+|uninstalling|Indicates that the partner agent will automatically be removed.  This occurs when there is no third-party-partner license but the third-party-partner agent was previously installed.|
```
Comment (Danielabom) [inline api-reference/beta/resources/cloudpcpartneragentinstallresult.md:41]:
```suggestion
|uninstalling|The partner agent will automatically be removed.  This occurs when there is no third-party-partner license but the third-party-partner agent was previously installed.|
```

**Review by Danielabom — CHANGES_REQUESTED** [review]
Hi @jannyma, thank you for working on this PR. Please review my comments. Thanks!

**Review by Danielabom — APPROVED** [review]
Thank you for addressing my comments @jannyma. Changes look good.

**Comment by jannyma** [comment]
> @jannyma
> 
> 1. please resolve comments in this PR.
> 2. Please also update the changelog/Microsoft.CloudManagedDesktop.json file later.
> 3. Please create a roadmap workitem and attach the workitem number in comments. Here is an [example](https://microsoftgraph.visualstudio.com/Roadmap/_workitems/edit/14460/). If no access, please do these two steps firstly.
> 
> * Request access via [Improved ADO Access - 1ES Wikirequest to join](https://www.1eswiki.com/wiki/Improved_ADO_Access) .
> * Join the Microsoft Graph Roadmap Users ([graphroadmap@microsoft.com](mailto:graphroadmap@microsoft.com)) security group on [idweb](https://idwebelements/GroupManagement.aspx?Group=graphroadmap&Operation=join) .

File: changelog/Microsoft.CloudManagedDesktop.json [hunk 21]
```diff
@@ -2573,6 +2573,32 @@
             "CreatedDateTime": "2022-08-24T02:53:28.597151Z",
             "WorkloadArea": "Devices and apps",
             "SubArea": "Cloud PC"
+        },
+        {
+            "ChangList": [
```
Comment (RuiHou105) [inline changelog/Microsoft.CloudManagedDesktop.json:2578]: For the changelog, you lack two: add new complex type `cloudPcPartnerAgentInstallResult`, new enum `cloudPcPartnerAgentInstallStatus`, `cloudPcPartnerAgentName`. Please add them here too

File: api-reference/beta/toc.yml [hunk 22]
```diff
@@ -10427,6 +10427,8 @@ items:
             href: api/manageddevice-setcloudpcreviewstatus.md
           - name: Troubleshoot
             href: api/cloudpc-troubleshoot.md
+          - name: RetryPartnerAgentInstallation
```
Comment (RuiHou105) [inline api-reference/beta/toc.yml:10430]:
```suggestion
          - name: Retry partner agent installation
```

File: api-reference/beta/api/cloudpc-retrypartneragentinstallation.md [hunk 23]
```diff
@@ -0,0 +1,84 @@
...
+doc_type: apiPageType
+---
+
+# cloudPC: retryPartnerAgentInstallation
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Attempts to re-install party partner agents in a Cloud PC failed to install.
```
Comment (RuiHou105) [inline api-reference/beta/api/cloudpc-retrypartneragentinstallation.md:16]:
```suggestion
Retry installation for the partner agents which failed to install on the Cloud PC. Service side will check which agent installation failed firstly and retry.
```

File: api-reference/beta/resources/cloudpc.md [hunk 24]
```diff
@@ -40,6 +40,7 @@ Represents a cloud-managed virtual desktop. This Cloud PC is also enrolled into
 |[List for user](../api/user-list-cloudpcs.md)|[cloudPC](../resources/cloudpc.md) collection|List the [cloudPC](../resources/cloudpc.md) devices that are attributed to the signed-in user.|
 |[Get launch info for user](../api/cloudpc-getcloudpclaunchinfo.md)|[cloudPCLaunchInfo](../resources/cloudpclaunchinfo.md)|Get the [cloudPCLaunchInfo](../resources/cloudpclaunchinfo.md) for the signed-in user.|
 |[Get connectivity history](../api/cloudpc-getcloudpcconnectivityhistory.md)|[cloudPcConnectivityEvent](../resources/cloudpcconnectivityevent.md) collection|Get the Cloud PC connectivity history.|
+|[Retry partner agent installation](../api/cloudpc-retrypartneragentinstallation.md)|None|Attempt to re-install party partner agents in a [cloudPC](../resources/cloudpc.md) which were failed to install.|
```
Comment (RuiHou105) [inline api-reference/beta/resources/cloudpc.md:43]: Please rephrase the description: Retry installation for the partner agents which failed to install on the Cloud PC.

**Review by RuiHou105 — APPROVED** [review]

**Comment by Danielabom** [comment]
@RuiHou105, @jannyma  is there a particular reason why you removed the "content review complete" label?

**Pull request merged**